app/controllers/insights_cloud/tasks_controller.rb

```diff
@@ -1,7 +1,12 @@
 module InsightsCloud
   class TasksController < ::ApplicationController
     def create
-      InsightsCloud::Async::InsightsFullSync.perform_now
+      InsightsCloud::Async::InsightsFullSync.perform_later
+      task = ForemanTasks::Task.where(action: 'InsightsCloud::Async::InsightsFullSync').reorder(started_at: :desc).first
```

Comment:
@adamruzicka is there a better way to get the task data here?

Reply:
`InsightsCloud::Async::InsightsFullSync.perform_later` returns a job object. You could take the task id from it. Something along the lines of this could work

```ruby
job = InsightsCloud::Async::InsightsFullSync.perform_later
ForemanTasks::Task.find_by(external_id: job.provider_job_id)
```